flycheck.el

```diff
@@ -6924,7 +6924,16 @@ See URL `http://www.erlang.org/'."
 
 (defun flycheck-rebar3-project-root (&optional _checker)
   "Return directory where =rebar.config= is located."
-  (locate-dominating-file buffer-file-name "rebar.config"))
+  (let* ((rebarconfig (locate-dominating-file buffer-file-name "rebar.config"))
+         (project-dir (if rebarconfig
+                          (file-name-as-directory
+                           (substring rebarconfig
+                                      0
+                                      (string-match
+                                       "\\(/_build/[^/]+/lib\\|/_checkouts\\)"
+                                       rebarconfig)))
```

Comment:
A `rebar3` project will check dependencies out into a `_build/<profile_name>/lib` folder. Your dependency will have `.erl` files, and you may want to open them in emacs to review what they're doing. If you do, `flycheck` will try and build that `rebar3` project, because it will satisfy the checker. 

Let's look at a real world example. https://github.com/joedevivo/chatterbox

Chatterbox has `src/*.erl` and we'll wan't those to work as the checker was already defined.

When I build chatterbox, it pulls dependencies into `_build/default/lib`, but also `_build/test/lib` which lets me include specific dependencies that I don't want bundled with the release, they're just there to test. If I open up any file in `_build/*/lib/*/src/*.erl`, it shouldn't try and compile that project on its own. That would wind up building `_build/<profile>/lib/<dependency>/_build/<profile>/lib/<dependency_of_the_first_depencency>` and so on. This regex will match places that rebar checks things out to, and helps us make sure we're only building the parent project.

`_checkouts` is a little different. Chatterbox is coupled to another repo I maintain: https://github.com/joedevivo/hpack. The development is done in tandem, and I can work on a master version of hpack by checking it out to `_checkouts`.

I hope this clarifies what that regex is doing here.